livesync_citadel/indico_livesync_citadel/backend.py

```diff
@@ -0,0 +1,111 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2002 - 2020 CERN
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+import json
+
+import requests
+from werkzeug.urls import url_join
+
+from indico.modules.attachments import Attachment
+from indico.modules.events import Event
+from indico.modules.events.contributions import Contribution
+from indico.modules.events.contributions.models.subcontributions import SubContribution
+from indico.modules.events.notes.models.notes import EventNote
+
+from indico_livesync import LiveSyncBackendBase, SimpleChange, Uploader
+from indico_livesync_citadel.models.search_id_map import EntryType, LiveSyncCitadelSearchAppIdMap
+from indico_livesync_citadel.schemas import (AttachmentSchema, ContributionSchema, EventNoteSchema, EventSchema,
+                                             SubContributionSchema)
+
+
+class LiveSyncCitadelUploaderError(Exception):
+    pass
+
+
+class LiveSyncCitadelUploader(Uploader):
+    def __init__(self, *args, **kwargs):
+        from indico_livesync_citadel.plugin import LiveSyncCitadelPlugin
+
+        super(LiveSyncCitadelUploader, self).__init__(*args, **kwargs)
+
+        self.search_app = LiveSyncCitadelPlugin.settings.get('search_backend_url')
+        self.tika_server = LiveSyncCitadelPlugin.settings.get('tika_server')
+        self.endpoint_url = url_join(self.search_app, 'api/records')
+        self.headers = {
+            'Content-Type': 'application/json',
+            'Accept': 'application/json',
+            'Authorization': 'Bearer {}'.format(LiveSyncCitadelPlugin.settings.get('search_backend_token'))
+        }
+
+    def get_jsondata(self, obj):
+        # TODO: should "schemas/indico" be another setting?
+        if isinstance(obj, Event):
+            event_data = EventSchema().dump(obj)
+            event_data['$schema'] = url_join(self.search_app, 'schemas/indico/events_v1.0.0.json')
+            return event_data, EntryType.event
+        elif isinstance(obj, Contribution):
+            contribution_data = ContributionSchema().dump(obj)
```

Comment:
Since we may serialize MANY objects in one go, I think this is one of the few cases where creating an instance of the schema on the module scope and using that instance all the time probably makes sense.